server/sonar-server/src/main/java/org/sonar/server/computation/ws/ActivityAction.java

```diff
@@ -17,144 +17,162 @@
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_COMPONENT_ID;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_COMPONENT_QUERY;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_MAX_EXECUTED_AT;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_MIN_SUBMITTED_AT;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_ONLY_CURRENTS;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_STATUS;
+import static org.sonarqube.ws.client.ce.CeWsParameters.PARAM_TYPE;
 
 public class ActivityAction implements CeWsAction {
-
-  private static final String PARAM_COMPONENT_UUID = "componentId";
-  private static final String PARAM_COMPONENT_QUERY = "componentQuery";
-  private static final String PARAM_TYPE = "type";
-  private static final String PARAM_STATUS = "status";
-  private static final String PARAM_ONLY_CURRENTS = "onlyCurrents";
-  private static final String PARAM_MIN_SUBMITTED_AT = "minSubmittedAt";
-  private static final String PARAM_MAX_EXECUTED_AT = "maxExecutedAt";
-
   private final UserSession userSession;
   private final DbClient dbClient;
   private final TaskFormatter formatter;
+  private final Set<String> taskTypes;
 
-  public ActivityAction(UserSession userSession, DbClient dbClient, TaskFormatter formatter) {
+  public ActivityAction(UserSession userSession, DbClient dbClient, TaskFormatter formatter, CeTaskProcessor[] taskProcessors) {
     this.userSession = userSession;
     this.dbClient = dbClient;
     this.formatter = formatter;
+
+    this.taskTypes = new LinkedHashSet<>();
+    for (CeTaskProcessor taskProcessor : taskProcessors) {
+      taskTypes.addAll(taskProcessor.getHandledCeTaskTypes());
+    }
   }
 
   @Override
   public void define(WebService.NewController controller) {
     WebService.NewAction action = controller.createAction("activity")
-      .setDescription(format("Search for past task executions. Requires the system administration permission, " +
-        "or project administration permission if %s is set.", PARAM_COMPONENT_UUID))
+      .setDescription(format("Search for tasks.<br> " +
+        "Requires the system administration permission, " +
+        "or project administration permission if %s is set.", PARAM_COMPONENT_ID))
       .setResponseExample(getClass().getResource("activity-example.json"))
       .setHandler(this)
       .setSince("5.2");
-    action.createParam(PARAM_COMPONENT_UUID)
-      .setDescription("Optional id of the component (project) to filter on")
+
+    action.createParam(PARAM_COMPONENT_ID)
+      .setDescription("Id of the component (project) to filter on")
       .setExampleValue(Uuids.UUID_EXAMPLE_03);
     action.createParam(PARAM_COMPONENT_QUERY)
-      .setDescription(format("Optional search by component name or key. Must not be set together with %s.", PARAM_COMPONENT_UUID))
+      // TODO add like on name and exact match on key
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1135)